Interface.C

```diff
@@ -234,13 +237,30 @@ void preciceAdapter::Interface::configureMesh(const fvMesh& mesh, const std::str
             // Assign the (x,y,z) locations to the vertices
             // TODO: Ensure consistent order when writing/reading
             for (int i = 0; i < faceNodes.size(); i++)
+            {
                 for (unsigned int d = 0; d < dim_; ++d)
+                {
                     vertices[verticesIndex++] = faceNodes[i][d];
+                }
+                verticesMap.emplace(std::make_tuple(faceNodes[i][0], faceNodes[i][1], faceNodes[i][2]), -1);
+            }
         }
 
         // Pass the mesh vertices information to preCICE
         precice_.setMeshVertices(meshName_, vertices, vertexIDs_);
 
+        // Build the map between OpenFOAM vertices and preCICE vertex IDs
+        verticesIndex = 0;
+        for (auto & key : verticesMap)
+        {
+            key.second = vertexIDs_[verticesIndex++];
```

Comment:
I had a quick look, but I think that this will not be safer. In the first loop, we iterate per buffer entry (x1, y1, z1, x2, y2, z2, ...) and need to set a triplet per vertex. In this loop, we iterate per vertex (v1, v2, v3).

Reply:
Ideally, we construct the complete map in one go only when `meshConnectivity_ == true`.

This version always builds the map, even if it is not required.

Reply:
Done